### PR TITLE
JERSEY-2879: Link HTTP Header: Fix for comma separated multiple header values

### PR DESCRIPTION
**Pull request by sfuhrm**
Fix for comma separated multiple header values as described in http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

See also examples http://tools.ietf.org/html/rfc5988#section-5.5

Issue for this pull: https://java.net/jira/browse/JERSEY-2879

### CODE REVIEW

**Comment by AdamLindenthal** [comment]
Jenkins, please test this patch.

**Comment by AdamLindenthal** [comment]
Hi Stephan,
thank you for your contribution!

Before we can review your pull request, we need you to sign [Oracle Contributor Agreement](http://www.oracle.com/technetwork/community/oca-486395.html). Please note, that it takes a while until the OCA is officially approved and listed on the page linked above (which is a relevant list for us).

Also to have the things nicely organised, every pull request should have a issue opened in our [bugtracker](https://java.net/jira/browse/JERSEY). And it is a good practice to place mutually links from github to JIRA and vice versa.

More information about contributing to Jersey: https://jersey.java.net/scm.html#/Submitting_Patches_and_Contribute_Code

Thanks,
Adam

File: core-common/src/main/java/org/glassfish/jersey/message/internal/InboundMessageContext.java [hunk 1]
```diff
@@ -653,7 +653,9 @@ public URI apply(String value) {
         try {
             Set<Link> result = new HashSet<Link>(links.size());
             for (String l : links) {
-                result.add(Link.valueOf(l));
+                for (String p : l.split(",")) {
```
Comment (mgajdos) [inline core-common/src/main/java/org/glassfish/jersey/message/internal/InboundMessageContext.java]: I am not sure that splitting the value is enough. URI itself can contain ``,`` which means that this code would produce invalid result.

**Comment by sfuhrm** [comment]
Fixed the "," handling in the URs as proposed, thanks for the hint.

**Comment by sfuhrm** [comment]
Can one of the admins verify this patch?

**Comment by mpotociar** [comment]
Jenkins, please test this patch.

**Comment by mpotociar** [comment]
Jenkins, please test this patch.

**Comment by mpotociar** [comment]
Please, try to rebase your change to the latest master. Seems that your current commits are based on a version that causes some copyright header check failures:
https://jersey-ci.ci.cloudbees.com/job/github-pull-request-verification/lastSuccessfulBuild/artifact/copyright.year.wrong.txt

**Comment by AdamLindenthal** [comment]
Jenkins, please test this patch.

**Comment by ruhkopf** [comment]
Hi,
Is there an update on this PR? Would be great to see proper RFC 5988 support as more and more APIs start using it (Github uses it, too, btw). Is this going to be merged/released soon?

File: core-common/src/main/java/org/glassfish/jersey/message/internal/InboundMessageContext.java [hunk 2]
```diff
@@ -59,6 +59,8 @@
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
+import java.util.StringTokenizer;
+import java.util.regex.Pattern;
```
Comment (AdamLindenthal) [inline core-common/src/main/java/org/glassfish/jersey/message/internal/InboundMessageContext.java]: "Pattern" import seems to be unused, please remove it.

**Comment by AdamLindenthal** [comment]
Hi Stephan,

just took another look at your patch and find a minor issue (see the inline comment). 
Otherwise it looks good to me. Please, remove the redundant import and the patch should be ready to merge.

Thanks,
Adam

**Comment by sfuhrm** [comment]
Hi Adam,

I removed the unneccesary Pattern import.
Would be great to be "ready to rock" now :)

Thanks,
Stephan

**Comment by AdamLindenthal** [comment]
Hi Stephan, I surely thing so. Let's give a while to Mr. Jenkins to confirm that :-)
Seriously, I looked at the commits, the build should pass, I just prefer to merge the verified build, if possible. 
There is a possibility, that the build will fail anyway, but presumably on known intermittent test issues, that are being worked on now (that happened recently with another PR). Don't worry, if that's the case, I will merge it anyway.

Adam

**Comment by AdamLindenthal** [comment]
Jenkins, please test this patch.

**Comment by AdamLindenthal** [comment]
Merged, thanks for contributing!

**Pull request merged**